judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/CallInterceptorUtil.java

```diff
@@ -0,0 +1,86 @@
+package hu.blackbelt.judo.runtime.core.dispatcher;
+
+import hu.blackbelt.judo.meta.asm.runtime.AsmModel;
+import org.eclipse.emf.ecore.EOperation;
+
+import java.util.concurrent.CompletableFuture;
+
+public class CallInterceptorUtil<P, R> {
+
+    private final Class<P> parameterType;
+    private final Class<R> returnType;
+
+    private final AsmModel asmModel;
+    private final EOperation operation;
+    private final OperationCallInterceptorProvider interceptorProvider;
+
+    @SuppressWarnings({"unchecked"})
+    public CallInterceptorUtil(Class<? super P> parameterType, Class<? super R> returnType, AsmModel asmModel,
+                               EOperation operation,
+                               OperationCallInterceptorProvider interceptorProvider) {
+        this.parameterType = (Class<P>) parameterType;
+        this.returnType = (Class<R>) returnType;
+        this.asmModel = asmModel;
+        this.operation = operation;
+        this.interceptorProvider = interceptorProvider;
+
+    }
+
+    public boolean isOriginalCalled() {
```

Comment:
Misleading method name, should be `shouldCallOriginal` because this is not an "after" helper.